assignments/this.js

```diff
@@ -13,14 +13,60 @@
 
 // code example for Window Binding
 
+window._ = {};
+
 // Principle 2
 
 // code example for Implicit Binding
 
+function foo() {
+    console.log(this.a);
+}
+
+const obj2 = {
+    a: 10,
+    foo: foo
+};
+
+const obj1 = {
+    a: 4,
+    obj2: obj2
+};
+
+obj1.obj2.foo();
+
 // Principle 3
 
 // code example for New Binding
 
+function foo(a) {
```

Comment:
please avoid using foo bar in your coding projects, naming multiple functions and variables the same name while also using var will cause you a lot of problems in your code and lead through a path of constant debugging. In this case your multiple foo and bars are clashing. Also, you should be putting these functions into variables so that they aren't firing off automatically when you don't want them to. If you start it off with just function then they are being told to fire off as soon as the page loads.